### PR TITLE
[Serverless] SLS-2094: Get KMS/Secrets Managers values from the Extension

### PR DESCRIPTION
**Pull request by IvanTopolcic**
### What does this PR do?

Automatically retrieve KMS/Secrets Manager values from AWS via the Datadog Extension.

### Motivation

Support dual shipping telemetry via the Datadog Lambda Extension with encrypted keys.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/serverless/api_key.go [hunk 1]
```diff
@@ -23,6 +23,12 @@ const encryptionContextKey = "LambdaFunctionName"
 // functionNameEnvVar is the environment variable that stores the function name.
 const functionNameEnvVar = "AWS_LAMBDA_FUNCTION_NAME"
 
+// kmsKeySuffix is the suffix of all environment variables which should be decrypted by KMS
+const kmsKeySuffix = "_KMS_KEY"
```
Comment (nhinsch) [inline cmd/serverless/api_key.go:27]: I don't think this suffix should be KMS key (see RFC comment for reasoning)

File: cmd/serverless/main.go [hunk 2]
```diff
@@ -154,10 +154,39 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 		log.Warn("An API Key has been set in multiple places:", strings.Join(apikeySetIn, ", "))
 	}
 
+	// first, generically parse all _SECRET_ARN and _KMS_KEY variables
```
[the remark this answers is not in the document]
Comment (IvanTopolcic) [reply cmd/serverless/main.go]: Done!

File: cmd/serverless/main.go [hunk 3]
```diff
@@ -154,10 +154,15 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 		log.Warn("An API Key has been set in multiple places:", strings.Join(apikeySetIn, ", "))
 	}
 
+	// first, generically parse all _SECRET_ARN and _KMS_KEY variable
+	for envKey, envVal := range getSecretEnvVars(os.Environ(), readAPIKeyFromSecretsManager, readAPIKeyFromKMS) {
```
Comment (maxday) [inline cmd/serverless/main.go:158]: maybe we could export this to a separate function a explicit name so we know directly what this code is doing (+ a comment?)

**Review by maxday — APPROVED** [review]
LGTM, just left a nit

File: cmd/serverless/main.go (hunk 3 above)
Comment (IvanTopolcic) [reply cmd/serverless/main.go:161]: Done 👍

**Pull request merged**